### PR TITLE
DOC-2130 feat(spark): read from tigergraph to Spark;

### PR DESCRIPTION
**Pull request by pingxieTG**
* Add page read-to-spark-dataframe to cover all read capabilities
* Revise load-from-spark-dataframe to fix some typos, minor mistakes
* Add `partials/spark` to include some sections that can be used by both spark read&write.

### CODE REVIEW

**Review by arun-TG-PM — APPROVED** [review]

File: modules/data-loading/partials/spark/prerequisite.adoc [hunk 1]
```diff
@@ -0,0 +1,46 @@
+=== Compatibility
+* TigerGraph 3.6.0 or higher.
+* Spark 3.2 or higher with Scala 2.12 and Scala 2.13.
+* JAVA 8 or higher.
+
+=== Download the JARs
+This connector can be downloaded from maven central repository:
```
Comment (victorleeTG) [inline modules/data-loading/partials/spark/prerequisite.adoc]:
```suggestion
This connector can be downloaded from the Maven central repository:
```

File: modules/data-loading/partials/spark/prerequisite.adoc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+https://central.sonatype.com/artifact/com.tigergraph/tigergraph-spark-connector/overview[Maven Central]
+it has 3 types of release format:
+
+
+
+* `tigergraph-spark-connector-<version>.jar`: The `JAR` file containing only the compiled classes of the connector, which does not include any dependencies.
+* `tigergraph-spark-connector-<version>-jar-with-dependencies.jar`: The `JAR` file that includes compiled classes, as well as all the dependencies.
+* `tigergraph-spark-connector-<version>.tar.gz`: The compressed `TAR` archive that includes `tigergraph-spark-connector-<version>.jar` and dependencies in separate `JAR` files.
+
+To use TigerGraph Spark connector in a Spark Shell:
```
Comment (victorleeTG) [inline modules/data-loading/partials/spark/prerequisite.adoc]:
```suggestion
To use the TigerGraph Spark connector in a Spark shell:
```

File: modules/data-loading/partials/spark/prerequisite.adoc [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+----
+
+[NOTE]
+====
+If you want to include TigerGraph Spark Connector in your Spark installation, add the `JAR` with dependencies to Spark's `jars` folder.
+====
+
+=== Configure Logging
+
+It’s highly recommended to set the log level to info to monitor the execution. There are two methods to enable logging for the connector.
```
Comment (victorleeTG) [inline modules/data-loading/partials/spark/prerequisite.adoc]:
```suggestion
We highly recommend setting the log level to `info` to monitor the execution. There are two methods to enable logging for the connector.
```

File: modules/data-loading/partials/spark/prerequisite.adoc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+==== Connector Default Logger
+The connector has a built-in logger based on _java.util.logging_. To activate it, you need to explicitly set the option `log.level` to 2, which enables info-level logging.
```
Comment (victorleeTG) [inline modules/data-loading/partials/spark/prerequisite.adoc]:
```suggestion
The connector has a built-in logger based on _java.util.logging_.
To activate it, set the option `log.level` to 2, which enables info-level logging.
```

File: modules/data-loading/partials/spark/prerequisite.adoc [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+Please refer to <<Common Options>> to learn more about `log.level` and `log.path` options.
```
Comment (victorleeTG) [inline modules/data-loading/partials/spark/prerequisite.adoc:38]: Not sure about this syntax  <<Common Options>>

File: modules/data-loading/pages/load-from-spark-dataframe.adoc [hunk 6]
```diff
@@ -19,71 +19,29 @@ and xref:tigergraph-server:data-loading:load-from-spark-dataframe.adoc#_load_dat
 
 This connector has multiple optimisations and is backward compatible with TigerGraph v3.6.0+. Although, the job level loading statistics is only available for v3.10.0+.
 
-[IMPORTANT]
-====
-The legacy xref:tigergraph-server:data-loading:spark-connection-via-jdbc-driver.adoc[Spark Connection Via JDBC Driver] will be deprecated. Please migrate to this new connector.
-====
+include::partial$spark/jdbc-deprecation.adoc[]
 
 == Prerequisite
```
Comment (victorleeTG) [inline modules/data-loading/pages/load-from-spark-dataframe.adoc:27]:
```suggestion
== Prerequisites
```

File: modules/data-loading/pages/load-from-spark-dataframe.adoc [hunk 7]
```diff
@@ -19,71 +19,29 @@ and xref:tigergraph-server:data-loading:load-from-spark-dataframe.adoc#_load_dat
 
 This connector has multiple optimisations and is backward compatible with TigerGraph v3.6.0+. Although, the job level loading statistics is only available for v3.10.0+.
```
Comment (victorleeTG) [inline modules/data-loading/pages/load-from-spark-dataframe.adoc]:
```suggestion
This connector has multiple optimizations and is backward compatible with TigerGraph v3.6.0+.

* Job level loading statistics is available in v3.10+.
* Support for reading data into Spark from TigerGraph is available in v4.1+.
```

**Review by victorleeTG — APPROVED** [review]

**Pull request merged**